File: app/src/main/java/com/example/smarthomecontrol/MainActivity.java
package com.example.smarthomecontrol;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Spinner;

/**
 * Screen 1 – Gesture Selection.
 * Displays a Spinner with 17 gestures. On selection, navigates to
 * ExpertVideoActivity (Screen 2) passing the gesture label as an extra.
 *
 * Privacy: No user face is captured or displayed on this screen.
 */
public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Spinner spinner = findViewById(R.id.gesture_spinner);

        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            private boolean isInitialSelection = true;

            @Override
            public void onItemSelected(AdapterView<?> parent, View view,
                                       int position, long id) {
                if (isInitialSelection) {
                    isInitialSelection = false;
                    return;
                }

                String displayName = parent.getItemAtPosition(position).toString();
                String gestureLabel = GestureConstants.GESTURE_MAP.get(displayName);

                Intent intent = new Intent(MainActivity.this, ExpertVideoActivity.class);
                intent.putExtra("EXTRA_GESTURE_LABEL", gestureLabel);
                startActivity(intent);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                // No action required
            }
        });
    }
}

File: app/src/main/java/com/example/smarthomecontrol/MainActivity.kt
package com.example.smarthomecontrol

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import com.example.smarthomecontrol.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var selectedGesture: Gesture? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupGestureSpinner()
        setupNextButton()
    }

    private fun setupGestureSpinner() {
        val gestureNames = Gesture.ALL_GESTURES.map { it.displayName }

        val adapter = ArrayAdapter(
            this,
            android.R.layout.simple_spinner_item,
            gestureNames
        ).apply {
            setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        }

        binding.gestureSpinner.adapter = adapter

        binding.gestureSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
                selectedGesture = Gesture.ALL_GESTURES[position]
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                selectedGesture = null
            }
        }

        // Select first item by default
        if (Gesture.ALL_GESTURES.isNotEmpty()) {
            binding.gestureSpinner.setSelection(0)
            selectedGesture = Gesture.ALL_GESTURES[0]
        }
    }

    private fun setupNextButton() {
        binding.nextButton.setOnClickListener {
            selectedGesture?.let { gesture ->
                val intent = Intent(this, VideoPlayActivity::class.java).apply {
                    putExtra(EXTRA_GESTURE_NAME, gesture.name)
                }
                startActivity(intent)
            }
        }
    }

    companion object {
        const val EXTRA_GESTURE_NAME = "extra_gesture_name"
    }
}
